### DISCUSSION TITLE
PV snapshot not getting restore across aws region

### DISCUSSION THREAD

**Discussion by deepanshutripathi** [vmware-tanzu/velero#4011] [Community support Q&A]
We have created the backup of EKS cluster (Command: velero create backup <backup_name>) using velero in souce region and this backup stored in the s3 bucket created in same region. Also EBS snapshot of pod's volume has been created by this command as well, in source region only.

While restoring the cluster from backup in destination region , we identify that all cluster resources has been restored including application pods as well, but Application pods were in "PENDING" state due to the  **"Volume node affinity conflict"** and on checking the PVC of pods saw below:

**Node Affinity:
  Required Terms:
    Term 0:        failure-domain.beta.kubernetes.io/zone in [eu-central-1c]
                   failure-domain.beta.kubernetes.io/region in [eu-central-1]**

I guess message "Volume node affinity conflict" clearly says that Snapshot in different region from one on which application pod has been scheduled hence failing.

Question: **So does this means Velero doesn't restore the PV snapshot across AWS region/Availability Zones, as in this case? If No, can please share any useful link or details?**

INFO: We have used TF code to create the EKS cluster in source and destination both region.

**Comment by dsu-igeek** [comment]
Currently snapshot restore in different regions is not supported by AWS and hence not supported by Velero.  eu-central-1c and eu-central-1, however, should be different availability zones, but not different regions.  Where was the snapshot originally taken?

**Reply by deepanshutripathi** [reply]
@dsu-igeek Thanks for your comment.

snapshot was originally taken on eu-central-1c AZ, during velero backup and using restore trying to restore it in us-east-1c.


**from your comment I can understand that velero supports cross AZs restoration within same region, but not in my case where I have cross AZ  across different regions ((i.e. source eu-central-1c and destination: us-east-1c)), is that right understanding?**

**Comment by jglick** [comment]
I implemented a prototype in https://github.com/vmware-tanzu/velero-plugin-for-aws/pull/90.
